src/System.Drawing.Common/tests/ImageTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace System.Drawing.Tests
+{
+    public class ImageTests
+    {
+        public static IEnumerable<object[]> InvalidBytes_TestData()
+        {
+            // IconTests.Ctor_InvalidBytesInStream_TestData an array of 2 objects, but this test only uses the
+            // 1st object.
+            foreach (object[] data in IconTests.Ctor_InvalidBytesInStream_TestData())
+            {
+                yield return new object[] { data[0] };
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(InvalidBytes_TestData))]
+        public void FromFile_InvalidBytes_ThrowsOutOfMemoryException(byte[] bytes)
+        {
+            using (var file = TempFile.Create(bytes))
+            {
+                Assert.Throws<OutOfMemoryException>(() => Image.FromFile(file.Path));
+                Assert.Throws<OutOfMemoryException>(() => Image.FromFile(file.Path, useEmbeddedColorManagement: true));
```

Comment:
Yeah. GdI+ status codes are pretty off sometimes. This doesn't affect memory but it's just what's returned.
In fact, you can see in the open source implementations of libgdiplus and wine that they throw this too explicitly and acknowledge that GDI+ doesn't really make sense with its errors sometimes